Automattic/wp-calypso
# NUX Signup: consolidate site topic submission code.

#### Changes proposed in this Pull Request

There are currently two ways of submitting a site topic during sign up: one via a query string and one via the site topic step. This PR attempts to de-duplicate the code for filling site topic info.

#### Parent PR
#28871 .

#### Testing instructions

1. Go through the improved onboarding flow from http://calypso.localhost:3000/start/onboarding and the site topic step should work as expected.
1. Accessing http://calypso.localhost:3000/start/onboarding?vertical=foo, `getState().signup` should show that the site topic, the dependency store, and the progress store are all filled in as expected.
